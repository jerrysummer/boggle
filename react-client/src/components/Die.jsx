import React from 'react';

const Die = ({ die, pos, handleDieSelect }) => (
  <div
    className={die.selected ? 'selected die' : 'die'}
    onClick={() => handleDieSelect(die.val, pos)}
  >
    <div className="die-circle">
      {die.val}
    </div>
  </div>
);

export default Die;
