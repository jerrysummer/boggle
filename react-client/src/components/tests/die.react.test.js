import React from 'react';
import renderer from 'react-test-renderer';
import Die from '../Die';

it('renders correctly when not selected', () => {
  const component = renderer
    .create(<Die die="{selected: false, val:A}" />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

it('renders correctly when selected', () => {
  const component = renderer
    .create(<Die die="{selected: true, val:A}" />)
    .toJSON();
  expect(component).toMatchSnapshot();
});


// it('changes the class when hovered', () => {
//   const component = renderer.create(<Die die="{selected: false, val:A}" />);
//   let tree = component.toJSON();
//   expect(tree).toMatchSnapshot();

//   // manually trigger the callback
//   tree.props.onMouseEnter();
//   // re-rendering
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();

//   // manually trigger the callback
//   tree.props.onMouseLeave();
//   // re-rendering
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });
