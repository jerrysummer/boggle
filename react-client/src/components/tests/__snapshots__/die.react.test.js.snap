// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when not selected 1`] = `
<div
  className="die"
  onClick={[Function]}
>
  <div
    className="die-circle"
  />
</div>
`;

exports[`renders correctly when selected 1`] = `
<div
  className="die"
  onClick={[Function]}
>
  <div
    className="die-circle"
  />
</div>
`;
